socketpair() is not in libc on illumos

_dbus_socketpair() not implemented on this OS
Unable to create reload pipe: _dbus_socketpair() not implemented on this OS

--- dbus-1.14.10/configure.ac.orig	2023-09-01 16:30:58.000000000 +0200
+++ dbus-1.14.10/configure.ac	2024-01-13 20:05:39.649204590 +0100
@@ -405,7 +405,6 @@
 setlocale
 setresuid
 setrlimit
-socketpair
 strtoll
 strtoull
 unsetenv
@@ -632,6 +631,11 @@
 AC_SEARCH_LIBS(socket,[socket network])
 AC_CHECK_FUNC(gethostbyname,,[AC_CHECK_LIB(nsl,gethostbyname)])
 
+AC_CHECK_FUNCS([socketpair], [], 
+  [AC_SEARCH_LIBS([socketpair], [socket],
+    [AC_DEFINE([HAVE_SOCKETPAIR], [1],
+      [Have socketpair function])])])
+
 AC_CHECK_HEADERS([syslog.h])
 if test "x$ac_cv_header_syslog_h" = "xyes"; then
   AC_CHECK_DECLS([LOG_PERROR], [], [], [[#include <syslog.h>]])
